registry/remote/auth/client.go

```diff
@@ -54,8 +54,15 @@ var maxResponseBytes int64 = 128 * 1024 // 128 KiB
 // See also ClientID.
 var defaultClientID = "oras-go"
 
+// CredentialFunc represents a function that resolves the credential for the
+// given registry (i.e. host:port).
+//
+// `EmptyCredential` is a valid return value and should not be considered as
+// an error.
+type CredentialFunc func(ctx context.Context, registry string) (Credential, error)
```

Comment:
Given the comment on the type, it might make more sense to name it `CredentialResolver`

Reply:
Yeah, but I feel that `CredentialFunc` matches more with the name `StaticCredential`
https://github.com/oras-project/oras-go/blob/deb17e4896db7b46f6e8a290ae64a74ae20c4894/registry/remote/auth/client.go#L65

and `credentials.Credential`.
https://github.com/oras-project/oras-go/blob/deb17e4896db7b46f6e8a290ae64a74ae20c4894/registry/remote/credentials/registry.go#L72

It's more concise and straightforward. 🤔